application/src/main/java/org/thingsboard/server/service/security/auth/oauth2/CustomOAuth2ClientMapper.java

```diff
@@ -57,6 +57,7 @@ private synchronized OAuth2User getOAuth2User(OAuth2AuthenticationToken token, S
         }
 
         RestTemplate restTemplate = restTemplateBuilder.build();
+        RestTemplateUtil.excludeJackson2XmlHttpMessageConverter(restTemplate);
```

Comment:
I don't get why the RestTemplate is created on each method call

Reply:
see lines 52-57.